### PR TITLE
restart policy for acdc

### PR DESCRIPTION
**Pull request by alecsavvy**
### Description
the acdc nodes will return a 503 on unhealthy status which will now trigger a restart of the service

### CODE REVIEW

File: discovery-provider/docker-compose.yml [hunk 1]
```diff
@@ -277,6 +277,17 @@ services:
         mode: non-blocking
         max-buffer-size: 100m
       driver: json-file
+    restart: unless-stopped
+    healthcheck:
+      test:         
+        [
+          "CMD-SHELL",
+          "curl --silent --fail http://localhost:8545/health || exit 1", # unhealthy nodes return a 503
```
Comment (isaacsolo) [inline discovery-provider/docker-compose.yml:285]: let's also check if a node is a signer before restarting

File: discovery-provider/chain/health.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+rpc_url = "http://localhost:8545"
+
+health_res = requests.get(health_url)
+health_json = health_res.json()
+
+healthy = True
+
+health_check_node = health_json.entries["node-health"]
+health_check_description = health_check_node.description
+health_check_healthy = health_json.status == "Unhealthy"
```
Comment (michellebrier) [inline discovery-provider/chain/health.py:13]: can we rename this to `health_check_unhealthy`? current name is kinda confusing

File: discovery-provider/chain/health.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+if health_check_healthy:
+    healthy = False
+
+not_producing_msg = "The node stopped producing blocks"
+if health_check_healthy and not_producing_msg in health_check_description:
```
Comment (michellebrier) [inline discovery-provider/chain/health.py:19]: what if a signer stops producing blocks? i think "The node stopped producing blocks" appears in the health check description then too

File: discovery-provider/chain/health.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    healthy = True
```
Comment (michellebrier) [inline discovery-provider/chain/health.py:20]: won't this [change](https://github.com/AudiusProject/audius-docker-compose/pull/219) result in both signers/non-signers reporting healthy?

**Pull request closed without merging**